java/code/src/com/suse/manager/webui/services/impl/SystemQuery.java

```diff
@@ -0,0 +1,498 @@
+/**
+ * Copyright (c) 2016 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.services.impl;
+
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.JsonElement;
+import com.redhat.rhn.domain.server.MinionServer;
+import com.suse.manager.reactor.SaltReactor;
+import com.suse.manager.virtualization.GuestDefinition;
+import com.suse.manager.webui.services.impl.runner.MgrK8sRunner;
+import com.suse.manager.webui.services.impl.runner.MgrUtilRunner;
+import com.suse.manager.webui.utils.gson.BootstrapParameters;
+import com.suse.manager.webui.utils.salt.custom.ScheduleMetadata;
+import com.suse.salt.netapi.calls.LocalAsyncResult;
+import com.suse.salt.netapi.calls.LocalCall;
+import com.suse.salt.netapi.calls.modules.SaltUtil;
+import com.suse.salt.netapi.calls.modules.State;
+import com.suse.salt.netapi.calls.modules.Zypper;
+import com.suse.salt.netapi.calls.runner.Jobs;
+import com.suse.salt.netapi.calls.wheel.Key;
+import com.suse.salt.netapi.datatypes.target.MinionList;
+import com.suse.salt.netapi.datatypes.target.Target;
+import com.suse.salt.netapi.errors.GenericError;
+import com.suse.salt.netapi.event.EventStream;
+import com.suse.salt.netapi.exception.SaltException;
+import com.suse.salt.netapi.results.Result;
+import com.suse.salt.netapi.results.SSHResult;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Interface containing methods for directly interacting and getting information from
+ * a system.
+ * Note: This interface should be split up further at some point.
+ */
+public interface SystemQuery {
```

Comment:
Is `SystemQuery` actually the most appropriate name we can find for this new interface? I find the name suitable for most of the functionality, but for some of the methods it doesn't really fit, like `generateSSHKey()` or the ones related to key management on the master. I don't currently have a better name for suggestion, just wanted to raise it for discussion. Maybe there could even be a separate interface for things that just operate locally on another iteration, is that what you mean with "split up further"?

Reply:
Yes that was the intention of splitting it further. Currently it contains all the methods we are using from `SaltService` but there is definitely different topics here that should go in their own interface. So far i'm thinking about splitting methods for general system information, salt specific methods like salt key handling etc, virtualization and a minimal low level salt interface for the salt backed implementations of those interfaces.